File: projects.py
class Project():
    ''' This class is for my projects
    Attributes:
        title (str):  Title of project.
        description (str): Description of project.
        details (str): Details of the project.
        url (str): Url link to website.
        source_code (str): Url link to github repo.
        tools (arr): Tools used in the development of project.
        thumbnail (str): Thumbnail preview image of project homepage.
        date (str): Date completed.
    '''

    def __init__(self, title, description, details, url, repo, tools, thumbnail, date):
        self.title = title
        self.description = description
        self.details = details
        self.url = url
        self.repo = repo
        toolString = ''
        i = 0
        while i < (len(tools) - 1):
            toolString += tools[i] + ", "
            i += 1
        toolString += "and " + tools[i] + '.'

        self.tools = toolString
        self.thumbnail = thumbnail
        self.date = date

File: js/projects.js


    function displayAll(){

        array = [{"description": "An image search microservice using the Google Custom Search API.", "title": "Image Search", "url": "https://imgsearch-ia.herokuapp.com", "repo": "https://github.com/TheArtilect/imageSearch", "date": "Oct. 2016", "tools": ["MongoDB", "Express", "Node", "Heroku"], "thumbnail": "ImageSearch.png"}, {"description": "A url shortener microservice.  Shortened url addresses redirect to the original url.  First project using MongoDB.", "title": "Url Shortener", "url": "https://u-short.herokuapp.com", "repo": "https://github.com/TheArtilect/urlShortener", "date": "Sept. 2016", "tools": ["MongoDB", "Express", "Node", "Heroku"], "thumbnail": "UrlShortener.png"}, {"description": "Fullstack javascript local weather app, utilizing Weather Underground API.  An improvement on previous local weather project.", "title": "Local Weather", "url": "https://weather-ia.herokuapp.com/", "repo": "https://github.com/TheArtilect/weather", "date": "Oct. 2016", "tools": ["Express", "Node", "Heroku"], "thumbnail": "weather.png"}]
        console.log("this worked")
        console.log(array)

    }

    $(document).ready(displayAll);
    $(document).on("page:load", displayAll)


